### PR TITLE
Add cashtag message entity type (close #19)

### LINKED ISSUE #19: Bot can't parse message with $CASHTAG
Hello!
If message contains a cashtag:
https://core.telegram.org/tdlib/docs/classtd_1_1td__api_1_1text_entity_type_cashtag.html
it will causes an error:
DecodeFailure "Error in $.result[0].message['reply_to_message'].entities[0].type: The key \"cashtag\" was not found".

### PR DESCRIPTION
**Pull request by kosc**
It solved the problem for me.

### CODE REVIEW

**Comment by swamp-agr** [comment]
@fizruk, I have no objections against this improvement.
May I merge it or, otherwise, could you please specify next steps?

**Comment by swamp-agr** [comment]
@kosc, could you please also update `haddock` to include `Cashtag` into documentation?

File: src/Telegram/Bot/API/Types.hs [hunk 1]
```diff
@@ -169,6 +169,7 @@ data MessageEntityType
   | MessageEntityPre
   | MessageEntityTextLink
   | MessageEntityTextMention
+  | MessageEntityCashtag
```
Comment (fizruk) [inline src/Telegram/Bot/API/Types.hs]:
```suggestion
  | MessageEntityCashtag -- ^ See <https://core.telegram.org/tdlib/docs/classtd_1_1td__api_1_1text_entity_type_cashtag.html>.
```

**Review by fizruk — APPROVED** [review]

**Comment by kosc** [comment]
> @kosc, could you please also update `haddock` to include `Cashtag` into documentation?

Done.

**Review by swamp-agr — APPROVED** [review]

**Pull request merged**